pennylane/beta/tapes/qnode.py

```diff
@@ -370,8 +396,16 @@ def __call__(self, *args, **kwargs):
         # construct the tape
         self.construct(args, kwargs)
 
+        if self._caching:
+            self.qtape._cache_execute = self._cache_execute
+
         # execute the tape
-        return self.qtape.execute(device=self.device)
+        res = self.qtape.execute(device=self.device)
+
+        if self._caching:
+            self._cache_execute = self.qtape._cache_execute

```

Comment:
Could only the qtape have the `_cache_execute` attribute? Since a QNode can access the attributes of a `qtape`

Reply:
Ah, this answers my question above! Probably a good idea to add a line comment, since Antal and I both independently had the same question

Reply:
If caching is on, can we avoid redundant tape constructions?

Reply:
I just had another quick try of: when caching is on, tape is only constructed first time.
This results in the tests related to mutability and classical processing to fail. For example, the cache is still being used when parameters are the same but the circuit differs. Although, I'm not sure if the problem is even deeper deeper, since isn't construction the place where the input arguments are fixed to the gate ops?